### PR TITLE
Fix PSScriptAnalyzer warnings

### PR DESCRIPTION
**Pull request by bergmeister**
To help fixing #3791 
Fixes a few hundred warnings of type PSPossibleIncorrectComparisonWithNull, PSAvoidUsingCmdletAliases (mostly for ForEach-Object, Where-Object, Select-Object) and some PSUseDeclaredVarsMoreThanAssignment warnings.
As part of this, it was also found that a test in ParameterBindingTests needed improvement because since the null comparison is now correct, the test failed initially.

### CODE REVIEW

File: test/tools/OpenCover/OpenCover.psm1 [hunk 1]
```diff
@@ -47,8 +47,8 @@ function Get-CodeCoverageChange($r1, $r2, [string[]]$ClassName)
 
     if ( $ClassName ) {
         foreach ( $Class in $ClassName ) {
-            $c1 = $r1.Assembly.ClassCoverage | ?{$_.ClassName -eq $Class }
-            $c2 = $r2.Assembly.ClassCoverage | ?{$_.ClassName -eq $Class }
+            $c1 = $r1.Assembly.ClassCoverage | Where-Object{$_.ClassName -eq $Class }
+            $c2 = $r2.Assembly.ClassCoverage | Where-Object{$_.ClassName -eq $Class }
```
Comment (iSazonov) [inline test/tools/OpenCover/OpenCover.psm1]: Please add space after `Where-Object` in both lines.

File: test/tools/OpenCover/OpenCover.psm1 [hunk 2]
```diff
@@ -136,7 +136,7 @@ function Get-CoverageData($xmlPath)
         Assembly = $assemblies
     }
     $CoverageData.PSTypeNames.Insert(0,"OpenCover.CoverageData")
-    Add-Member -InputObject $CoverageData -MemberType ScriptMethod -Name GetClassCoverage -Value { param ( $name ) $this.assembly.classcoverage | ?{$_.classname -match $name } }
+    Add-Member -InputObject $CoverageData -MemberType ScriptMethod -Name GetClassCoverage -Value { param ( $name ) $this.assembly.classcoverage | Where-Object{$_.classname -match $name } }
```
Comment (iSazonov) [inline test/tools/OpenCover/OpenCover.psm1]: Please add space after `Where-Object`.

**Review by iSazonov — COMMENTED** [review]
As requested in #3791 we should add meta test with scriptanalyzer.

**Comment by bergmeister** [comment]
Thanks @iSazonov for taking such a detailed look and spotting the accidental mistakes that I made when going through the hundreds of refactorings. I have followed all your suggestions and submitted the fix as you can see above. The total number of warnings seems to have gone down by about 350.

File: src/System.Management.Automation/engine/Types_Ps1Xml.cs [hunk 3]
```diff
@@ -232,10 +232,10 @@ public static IEnumerable<TypeData> Get()
             var td24 = new TypeData(@"System.Management.ManagementObject#root\cimv2\Win32_PingStatus", true);
             td24.Members.Add("IPV4Address",
                 new ScriptPropertyData(@"IPV4Address", GetScriptBlock(@"$iphost = [System.Net.Dns]::GetHostEntry($this.address)
-          $iphost.AddressList | Where-Object{ $_.AddressFamily -eq [System.Net.Sockets.AddressFamily]::InterNetwork } | Select-Object -first 1"), null));
+          $iphost.AddressList | Where-Object{  $_.AddressFamily -eq [System.Net.Sockets.AddressFamily]::InterNetwork } | Select-Object -first 1"), null));
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/Types_Ps1Xml.cs]: Please remove extra space before `$_` and add space after `Where-Object` and before `{`.

The same typo is below.

File: test/powershell/Modules/Microsoft.PowerShell.Security/AclCmdlets.Tests.ps1 [hunk 4]
```diff
@@ -12,7 +12,7 @@ Describe "Acl cmdlets are available and operate properly" -Tag CI {
         { $acl | Set-Acl $directory } | should not throw
 
         $newacl = get-acl $directory
-        $newrule = $newacl.Access | Where-Object{ $accessrule.FileSystemRights -eq $_.FileSystemRights -and $accessrule.AccessControlType -eq $_.AccessControlType -and $accessrule.IdentityReference -eq $_.IdentityReference }
+        $newrule = $newacl.Access | Where-Object{  $accessrule.FileSystemRights -eq $_.FileSystemRights -and $accessrule.AccessControlType -eq $_.AccessControlType -and $accessrule.IdentityReference -eq $_.IdentityReference }
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Security/AclCmdlets.Tests.ps1]: Please add space after `Where-Object` and remove extra space before `$accessrule`.

File: test/powershell/Modules/Microsoft.PowerShell.Utility/Implicit.Remoting.Tests.ps1 [hunk 5]
```diff
@@ -808,7 +808,7 @@ Describe "Import-PSSession functional tests" -tags "Feature" {
         }
 
         It "Temporary module should be automatically removed after runspace is closed" -Skip:$skipTest {
-            ($null -eq (Get-Module | Where-Object { $_.Path -eq $module.Path })) | Should Be $true
+             (Get-Module | Where-Object { $_.Path -eq $module.Path }) | Should BeNullOrEmpty
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/Implicit.Remoting.Tests.ps1]: Indentation.

File: test/powershell/Modules/PackageManagement/PackageManagement.Tests.ps1 [hunk 6]
```diff
@@ -32,9 +32,9 @@ Describe "PackageManagement Acceptance Test" -Tags "Feature" {
 
         $gpp = Get-PackageProvider
 
-        $gpp | Where-Object{ $_.name -eq "NuGet" } | should not BeNullOrEmpty
+        $gpp | Where-Object{  $_.name -eq "NuGet" } | should not BeNullOrEmpty
```
Comment (iSazonov) [inline test/powershell/Modules/PackageManagement/PackageManagement.Tests.ps1]: Please remove extra space.

File: test/powershell/Modules/PackageManagement/PackageManagement.Tests.ps1 [hunk 7]
```diff
@@ -32,9 +32,9 @@ Describe "PackageManagement Acceptance Test" -Tags "Feature" {
...
 
-        $gpp | Where-Object{ $_.name -eq "PowerShellGet" } | should not BeNullOrEmpty
+        $gpp | Where-Object{  $_.name -eq "PowerShellGet" } | should not BeNullOrEmpty
```
Comment (iSazonov) [inline test/powershell/Modules/PackageManagement/PackageManagement.Tests.ps1]: Please remove extra space.

File: test/tools/OpenCover/OpenCover.psm1 [hunk 8]
```diff
@@ -136,7 +136,7 @@ function Get-CoverageData($xmlPath)
         Assembly = $assemblies
     }
     $CoverageData.PSTypeNames.Insert(0,"OpenCover.CoverageData")
-    Add-Member -InputObject $CoverageData -MemberType ScriptMethod -Name GetClassCoverage -Value { param ( $name ) $this.assembly.classcoverage | Where-Object{$_.classname -match $name } }
+    Add-Member -InputObject $CoverageData -MemberType ScriptMethod -Name GetClassCoverage -Value { param ( $name ) $this.assembly.classcoverage | Where-Object{ $_.classname -match $name } }
```
Comment (iSazonov) [inline test/tools/OpenCover/OpenCover.psm1]: Please remove extra space and add space after `Where-Object`.

File: src/Modules/Windows-Full/Microsoft.PowerShell.ODataUtils/Microsoft.PowerShell.ODataV4Adapter.ps1 [hunk 9]
```diff
@@ -1495,7 +1495,7 @@ function SaveCDXMLSingletonCmdlets
 	{
 		$xmlWriter = SaveCDXMLHeader $xmlWriter $Uri $singletonName $singletonName $CmdletAdapter
 
-        if ($null -eq $associatedEntityType.BaseTyp -and $null -ne $associatedEntityType.BaseTypeStr -and $associatedEntityType.BaseTypeStr -ne '')
+        if ($null -eq $associatedEntityType.BaseType -and $null -ne $associatedEntityType.BaseTypeStr -and $associatedEntityType.BaseTypeStr -ne '')
```
Comment (lzybkr) [inline src/Modules/Windows-Full/Microsoft.PowerShell.ODataUtils/Microsoft.PowerShell.ODataV4Adapter.ps1]: @anmenaga - this looks like a bug that should get fixed in Windows PowerShell.

**Comment by lzybkr** [comment]
I'm seeing different sorts of changes in a single commit - and with this many changes, it would be easier and safer to keep each refactoring as a single commit.

**Comment by iSazonov** [comment]
+1 for multiple commits. Maybe:
1. Build.psm1
2. Rest tool scripts
3. Test scripts
4. Rest files.

**Comment by bergmeister** [comment]
@iSazonov I have fixed your requested changes about the fine-tuning of the spacing. When I started out doing that, I initially wanted to fix only most PSPossibleIncorrectComparisonWithNull violations but then I came across lots of other things to be improved like e.g. the aliases. Because the contribution guide [here](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#lifecycle-of-a-pull-request) recommended to have just one commit per change, I thought I'd try to put it all into one commit for the pull request. I am not sure how one would change the  commit history now that the commits are already pushed to the remote. Sorry, maybe I should have rather checked with you before.

**Comment by iSazonov** [comment]
You would rebase and split - see https://stackoverflow.com/questions/6217156/break-a-previous-commit-into-multiple-commits

File: src/Modules/Windows-Full/Microsoft.PowerShell.ODataUtils/Microsoft.PowerShell.ODataV4Adapter.ps1 (hunk 9 above)
Comment (bergmeister) [inline src/Modules/Windows-Full/Microsoft.PowerShell.ODataUtils/Microsoft.PowerShell.ODataV4Adapter.ps1]: @lzybkr : I'm not sure what you mean but the missing 'e' on 'BaseType' was an accidental deletion that happened in the first commit that I corrected

Comment (lzybkr) [inline src/Modules/Windows-Full/Microsoft.PowerShell.ODataUtils/Microsoft.PowerShell.ODataV4Adapter.ps1]: I see - this is a good example of why I'm not a fan of mixing minor changes in one big commit, because I do sometimes review commits by themselves instead of the full PR.

**Comment by bergmeister** [comment]
@iSazonov OK, thanks. I did not feel confident/adventurous enough to rebase commits that are already pushed to the remote, so I have created a new branch, pulled in my commit changes and made then the 4 separate commits as requested by you. I have created the new PR #4213 for that, so feel free to close this one.
